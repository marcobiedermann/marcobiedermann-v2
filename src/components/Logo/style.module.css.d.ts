/* eslint-disable import/prefer-default-export */

export const logo: string;
