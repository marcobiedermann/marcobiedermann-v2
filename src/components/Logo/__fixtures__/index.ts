const project = {
  id: '1',
  logo: {
    file: {
      url: 'https://via.placeholder.com/150x50',
    },
    title: 'Logo Title',
  },
};

export default project;
