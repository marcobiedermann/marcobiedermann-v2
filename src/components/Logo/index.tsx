import classNames from 'classnames';
import { Link } from 'gatsby';
import React, { FC } from 'react';
import * as styles from './style.module.css';

export interface LogoProps {
  className?: string;
  id: string;
  logo: {
    file: {
      url: string;
    };
    title: string;
  };
  slug?: string;
}

export const Logo: FC<LogoProps> = (props) => {
  const { className, logo, slug } = props;

  return (
    <Link to={`/projects/${slug}`} className={classNames(className, styles.logo)}>
      {logo && <img src={logo.file.url} alt={logo.title} />}
    </Link>
  );
};

export default Logo;
