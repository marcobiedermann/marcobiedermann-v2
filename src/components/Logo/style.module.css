.logo {
  filter: grayscale(1);
  transition: filter 0.5s;
}

.logo:focus,
.logo:hover {
  filter: grayscale(0);
}
