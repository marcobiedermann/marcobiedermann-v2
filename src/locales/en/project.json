{
  "agency": "Agency",
  "client": "Client",
  "colors": {
    "title": "Colors"
  },
  "homepage": {
    "title": "Homepage"
  },
  "mobile": {
    "description": "With a small display comes great responsibilities",
    "title": "Mobile View"
  },
  "more work": "See more Work",
  "tablet": {
    "title": "Tablet View"
  },
  "role": "Role",
  "responsive": {
    "description": "A website that displays nicely on desktop and on your phone and tablet",
    "title": "Responsive Design"
  },
  "typography": {
    "title": "Typography"
  },
  "website": "Visit Website"
}
