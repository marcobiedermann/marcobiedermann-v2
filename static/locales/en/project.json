{
  "agency": "Agency",
  "client": "Client",
  "colors": {
    "title": "Colors"
  },
  "homepage": {
    "title": "Homepage"
  },
  "mobile": {
    "description": "With a small display comes great responsibilities",
    "title": "Mobile View"
  },
  "more work": "See more Work",
  "tablet": {
    "title": "Tablet View"
  },
  "role": "Role",
  "responsive": "Responsive Design",
  "typography": {
    "title": "Typography"
  }
}
