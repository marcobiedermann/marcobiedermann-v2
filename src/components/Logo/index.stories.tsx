/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Logo from '.';
import logo from './__fixtures__';

storiesOf('Logo', module).add('default', () => <Logo {...logo} />);
