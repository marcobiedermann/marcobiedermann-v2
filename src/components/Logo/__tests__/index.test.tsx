/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Logo from '..';
import project from '../__fixtures__';

describe('Logo component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Logo {...project} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
