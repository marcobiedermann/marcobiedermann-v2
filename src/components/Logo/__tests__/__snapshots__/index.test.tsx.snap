// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render correctly 1`] = `
<DocumentFragment>
  <a
    class="logo"
    href="/projects/undefined"
  >
    <img
      alt="Logo Title"
      src="https://via.placeholder.com/150x50"
    />
  </a>
</DocumentFragment>
`;
